FTDI driver support

filipleple opened the pull request.

filipleple committed:
payloads/external/edk2/Makefile: add edk2-platforms/Drivers path

Signed-off-by: Filip Lewiński <filip.lewinski@3mdeb.com>

filipleple committed:
configs/config.novacustom_*: bump edk2* revs for FtdiUsbSerialDxe support

Signed-off-by: Filip Lewiński <filip.lewinski@3mdeb.com>

filipleple committed:
payloads/external/edk2/Kconfig: add CONFIG_EDK2_FTDI_USB_UART_SUPPORT

Signed-off-by: Filip Lewiński <filip.lewinski@3mdeb.com>

filipleple committed:
configs/config.novacustom_*: add CONFIG_EDK2_FTDI_USB_UART_SUPPORT=y

Signed-off-by: Filip Lewiński <filip.lewinski@3mdeb.com>

filipleple committed:
configs/config.novacustom*: bump edk2-platforms revision

Signed-off-by: Filip Lewiński <filip.lewinski@3mdeb.com>

filipleple committed:
.gitignore: add uefiextract

UEFIExtract gets downloaded when building for laptops with ./build.sh as of
recent. Adding it to .gitignore for convenience.

Signed-off-by: Filip Lewiński <filip.lewinski@3mdeb.com>

miczyg1 reviewed (dismissed):
Since you introduce a new Kconfig option for EDK2 you have to bump edk2 and edk2-platforms revisions in every config. Otherwise, this new option will remain broken on the platform which do not update the edk2 and edk2-platforms. @filipleple

filipleple committed:
configs/: update edk2 and edk2-platforms revisions

Accounts for the itroduction of the CONFIG_EDK2_FTDI_USB_UART_SUPPORT=y
Kconfig option

Signed-off-by: Filip Lewiński <filip.lewinski@3mdeb.com>

filipleple commented:
Right, thanks, updated for all platforms

mkopec reviewed (approved):
I've tested a binary for NS5x TGL from CI and it seems to work as expected :+1:
